cilium-cli/connectivity/check/deployment.go

```diff
@@ -1179,6 +1135,255 @@ func (ct *ConnectivityTest) deploy(ctx context.Context) error {
+			Replicas: replicas,
+			Labels:   map[string]string{"app": appLabel},
+			Command: []string{
+				"tcd-client",
+				"--dispatch-interval", ct.params.ConnDisruptDispatchInterval.String(),
+				address,
+			},
+			ReadinessProbe: readinessProbe,
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{corev1.ResourceCPU: *resource.NewMilliQuantity(100, resource.DecimalSI)},
+			},
+		}
+		if isExternal {
+			param.NodeSelector = map[string]string{"cilium.io/no-schedule": "true"}
+			param.HostNetwork = true
+			param.Tolerations = []corev1.Toleration{
+				{Operator: corev1.TolerationOpExists},
+			}
+		}
+		testConnDisruptClientDeployment := newDeployment(param)
+
+		_, err = ct.clients.dst.CreateServiceAccount(ctx, ct.params.TestNamespace, k8s.NewServiceAccount(deployName), metav1.CreateOptions{})
+		if err != nil {
+			return fmt.Errorf("unable to create service account %s: %w", deployName, err)
+		}
+		_, err = ct.clients.dst.CreateDeployment(ctx, ct.params.TestNamespace, testConnDisruptClientDeployment, metav1.CreateOptions{})
+		if err != nil {
+			return fmt.Errorf("unable to create deployment %s: %w", testConnDisruptClientDeployment, err)
+		}
+	}
+
+	return err
+}
+
+func (ct *ConnectivityTest) createTestConnDisruptClientDeploymentForNSTraffic(ctx context.Context) error {
+	nodes, err := ct.getBackendNodeAndNonBackendNode(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, n := range nodes {
+		for _, client := range ct.Clients() {
+			svc, err := client.GetService(ctx, ct.params.TestNamespace, testConnDisruptNSTrafficServiceName, metav1.GetOptions{})
+			if err != nil {
+				return fmt.Errorf("unable to get service %s: %w", testConnDisruptNSTrafficServiceName, err)
+			}
+
+			var errs error
+			np := uint32(svc.Spec.Ports[0].NodePort)
+			addrs := slices.Clone(n.node.Status.Addresses)
+			hasNetworkPolicies, err := ct.hasNetworkPolicies(ctx)
+			if err != nil {
+				return fmt.Errorf("failed to check if any netpol exists: %w", err)
+			}
+			ct.ForEachIPFamily(hasNetworkPolicies, func(family features.IPFamily) {
+				for _, addr := range addrs {
+					if features.GetIPFamily(addr.Address) != family {
+						continue
+					}
+
+					// On GKE ExternalIP is not reachable from inside a cluster
+					if addr.Type == corev1.NodeExternalIP {
+						if f, ok := ct.Feature(features.Flavor); ok && f.Enabled && f.Mode == "gke" {
+							continue
+						}
+					}
+
+					addrFormat := "%s:%d"
+					if family == features.IPFamilyV6 {
+						addrFormat = "[%s]:%d"
+					}
+					deployName := fmt.Sprintf("%s-%s-%s-%s", testConnDisruptClientNSTrafficDeploymentName, n.nodeType, family, strings.ToLower(string(addr.Type)))
+					if err := ct.createTestConnDisruptClientDeployment(ctx,
+						deployName,
+						KindTestConnDisruptNSTraffic,
+						fmt.Sprintf("test-conn-disrupt-client-%s-%s-%s", n.nodeType, family, strings.ToLower(string(addr.Type))),
+						fmt.Sprintf(addrFormat, addr.Address, np),
```

Comment:
Nit: Instead of specifying format manually, you can do `netip.AddrPortFrom(netip.MustParseAddr(addr.Address), np).String()`.

https://go.dev/play/p/bxawMNkZFZx

Reply:
Fixed, ty!